src/external/spglib.jl

```diff
@@ -162,3 +162,82 @@ function spglib_get_stabilized_reciprocal_mesh(kgrid_size, rotations::Vector;
 
     return n_kpts, Int.(mapping), [Vec3{Int}(grid_address[:, i]) for i in 1:nkpt]
 end
+
+
+struct SpglibDataset
+    spacegroup_number::Cint
+    hall_number::Cint
+    international_symbol::NTuple{11, Cchar}
+    hall_symbol::NTuple{17, Cchar}
+    choice::NTuple{6, Cchar}
+    transformation_matrix::NTuple{9, Cdouble}
+    origin_shift::NTuple{3, Cdouble}
+    n_operations::Cint
+    rotations::Ptr{NTuple{9, Cint}}
+    translations::Ptr{NTuple{3, Cdouble}}	
+    n_atoms::Cint
+    wyckoffs::Ptr{Cint}
+    site_symmetry_symbols::Ptr{NTuple{7, Cchar}}
+    equivalent_atoms::Ptr{Cint}
+    crystallographic_orbits::Ptr{Cint}
+    primitive_lattice::NTuple{9, Cdouble}
+    mapping_to_primitive::Ptr{Cint}
+    n_std_atoms::Cint
+    std_lattice::NTuple{9, Cdouble}
+    std_types::Ptr{Cint}
+    std_positions::NTuple{3, Cdouble}
+    std_rotation_matrix::NTuple{9, Cdouble}
+    std_mapping_to_primitive::Ptr{Cint}
+    pointgroup_symbol::NTuple{6, Cchar}
+end
+
+function unload_spglib_dataset(dataset)
```

Comment:
I get your idea to abstract the C interface completely, but I think
- this is a lot of boilerplate code we don't really need
- to truly abstract the interface you also need to take of unit conversions etc, which is quite a pain and not worth it in my opinion.

I would suggest to consider the dataset internal from the point of view of spglib.jl, i.e. inside spglib.jl the c-types and spglib conventions are ok to be used, so you don't need to convert to julia types. The things we need outside e.g. `spglib_get_spacegroup` for those we do the conversion to Julia types selectively, but only there.

Reply:
Okay! Once the memory issue is resolved I will fix this too